packages/zoe/src/typeGuards.js

```diff
@@ -79,3 +121,231 @@ export const InvitationElementShape = M.splitRecord({
+  ),
+  makeNoEscrowSeat: M.call(
+    AmountKeywordRecordShape,
+    ProposalShape,
+    M.remotable('ExitObj'),
+    SeatShape,
+  ).returns({ userSeat: SeatShape }),
+  replaceAllocations: M.call(SeatHandleAllocations).returns(),
+  exitAllSeats: M.call(M.any()).returns(),
+  failAllSeats: M.call(M.any()).returns(),
+  exitSeat: M.call(SeatShape, M.any()).returns(),
+  failSeat: M.call(SeatShape, M.any()).returns(),
+  stopAcceptingOffers: M.call().returns(),
+  setOfferFilter: M.call(M.arrayOf(M.string())).returns(),
+  getOfferFilter: M.call().returns(M.arrayOf(M.string())),
+  getExitSubscriber: M.call(SeatShape).returns(SubscriberShape),
+});
+
+export const InstanceStorageManagerIKit = harden({
+  instanceStorageManager: M.interface('InstanceStorageManager', {
+    getTerms: M.call().returns(M.split(TermsShape, M.record())),
+    getIssuers: M.call().returns(IssuerKeywordRecordShape),
+    getBrands: M.call().returns(BrandKeywordRecordShape),
+    getInstallationForInstance: M.call().returns(InstallationShape),
+    getInvitationIssuer: M.call().returns(IssuerShape),
+
+    saveIssuer: M.call(IssuerShape, M.string()).returns(M.promise()),
+    makeZoeMint: M.call(KeywordShape)
+      .optional(AssetKindShape, DisplayInfoShape, M.pattern())
+      .returns(M.or(ZoeMintI, M.remotable('zoeMint'), M.promise())),
+    registerFeeMint: M.call(KeywordShape, M.remotable('feeMintAccess')).returns(
+      M.remotable('feeMint'),
+    ),
+    getInstanceRecord: M.call().returns(InstanceRecordShape),
+    getIssuerRecords: M.call().returns(M.arrayOf(IssuerRecordShape)),
+    getWithdrawFacet: M.call().returns(M.remotable('WithdrawFacet')),
+    initInstanceAdmin: M.call(
+      InstanceHandleShape,
+      M.remotable('instanceAdmin'),
+    ).returns(M.promise()),
+    deleteInstanceAdmin: M.call(InstanceAdminI).returns(),
+    makeInvitation: M.call(InvitationHandleShape, M.string())
+      .optional(M.any(), M.pattern())
+      .returns(M.promise()),
+    getRoot: M.call().returns(M.any()),
+    getAdminNode: M.call().returns(M.remotable('adminNode')),
+  }),
+  withdrawFacet: M.interface('WithdrawFacet', {
+    withdrawPayments: M.call(AmountKeywordRecordShape).returns(
+      PaymentPKeywordRecordShape,
+    ),
+  }),
+});
+
+// In `SwingSet/src/types-external.js`, it's typed as `*`.
+const BundleCapShape = M.any();
+
+export const ZoeStorageManagerIKit = harden({
+  zoeServiceDataAccess: M.interface('ZoeService dataAccess', {
+    getTerms: M.call(InstanceHandleShape).returns(
+      M.split(TermsShape, M.record()),
+    ),
+    getIssuers: M.call(InstanceHandleShape).returns(IssuerKeywordRecordShape),
+    getBrands: M.call(InstanceHandleShape).returns(BrandKeywordRecordShape),
+    getInstallationForInstance: M.call(InstanceHandleShape).returns(
+      M.or(M.promise(), M.remotable('Installation')),
+    ),
+    getInvitationIssuer: M.call().returns(IssuerShape),
+
+    getBundleIDFromInstallation: M.call(M.any()).returns(M.promise()),
```

Comment:
Does this make sense here?

```suggestion
    getBundleIDFromInstallation: M.call(InstallationShape).returns(M.promise()),
```

Reply:
yep. it's only an allegedInstallation, but that's what it's supposed to be.

Reply:
What about line 325?

Reply:
These are the same now.